packages/plugin-pre/src/PreDomRenderer.ts

```diff
@@ -0,0 +1,53 @@
+import { AbstractRenderer } from '../../plugin-renderer/src/AbstractRenderer';
+import { DomRenderingEngine } from '../../plugin-dom/src/DomRenderingEngine';
+import { PreNode } from './PreNode';
+
+export class PreDomRenderer extends AbstractRenderer<Node[]> {
+    static id = 'dom';
+    engine: DomRenderingEngine;
+    predicate = PreNode;
+
+    /**
+     * Render the VNode to the given format.
+     */
+    async render(node: PreNode): Promise<Node[]> {
+        const pre = document.createElement('pre');
+        this.engine.renderAttributes(node.attributes, pre);
+        for (const child of node.children()) {
+            let domChildren = await this.renderChild(child);
+            // TODO: prevent CharNode from rendering spaces as nbsp.
```

Comment:
A possible way would to load a special parser which predicate recognizes CharNode inside PreNode but I fear that would lead to quite the duplication I think...

Reply:
https://github.com/odoo-dev/jabberwock/pull/162/commits/05495ed7e1359a8b8f5fd12c85e4da15b23f2e5e